import-automation/progress-dashboard-rest/test/test_progress_log_list.py

```diff
@@ -0,0 +1,203 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Tests for progress_log_list.py.
+"""
+
+import unittest
+from unittest import mock
+
+from test import utils
+from app.resource import system_run_list
+from app.resource import progress_log_list
+from app.resource import import_attempt_list
+from app.model import system_run_model
+from app.model import import_attempt_model
+from app.model import progress_log_model
+
+_ATTEMPT = import_attempt_model.ImportAttemptModel
+_RUN = system_run_model.SystemRunModel
+_LOG = progress_log_model.ProgressLogModel
+
+
+class ProgressLogListTest(unittest.TestCase):
+    """Tests for ProgressLogList."""
+
+    @classmethod
+    def setUpClass(cls):
+        cls.emulator = utils.start_emulator()
+
+    @classmethod
+    def tearDownClass(cls):
+        utils.terminate_emulator(cls.emulator)
+
+    @mock.patch('app.utils.create_datastore_client',
+                utils.create_test_datastore_client)
+    @mock.patch('app.service.log_message_manager.LogMessageManager',
+                utils.LogMessageManagerMock)
+    def setUp(self):
+        """Injects several system runs and import attempts to the database
+        before every test. New progress logs will be linked to these entities.
+        """
+        self.resource = progress_log_list.ProgressLogList()
+        run_list_resource = system_run_list.SystemRunList()
+        attempt_list_resource = import_attempt_list.ImportAttemptList()
+        run_list_resource.database.client = self.resource.client
+        attempt_list_resource.database.client = self.resource.client
+        attempt_list_resource.run_database.client = self.resource.client
+
+        runs = [
+            {_RUN.branch_name: 'test-branch', _RUN.pr_number: 0},
+            {_RUN.branch_name: 'prod-branch', _RUN.pr_number: 1}
+        ]
+        self.runs = utils.ingest_system_runs(run_list_resource, runs)
+
+        # Linked to the first run
+        attempts = [
+            {_ATTEMPT.import_name: 'cpi-u'},
+            {_ATTEMPT.import_name: 'cpi-w'},
+        ]
+        self.attempts = utils.ingest_import_attempts(
+            run_list_resource, attempt_list_resource,
+            attempts, system_run=self.runs[0])
+
+    @mock.patch(utils.PARSE_ARGS)
+    def test_level_not_allowed(self, parse_args):
+        """Tests that POSTing a progress log with an undefined log level returns
+        FORBIDDEN."""
+        parse_args.return_value = {
+            _LOG.level: 'not-exist',
```

Comment:
for log levels, have you considered using the predefined levels in Absl logging or even the python standard logging?  https://abseil.io/docs/python/guides/logging, https://legacy.python.org/dev/peps/pep-0282/

Reply:
switched to python standard logging levels